src/controllers/clients.controller.ts

```diff
@@ -1,20 +1,49 @@
-import { fetchClients } from "../models/users.models";
+import { fetchClients, getClient } from "../models/clients.models";
 import ApiError from "../utils/ApiError";
 import httpStatus from "http-status";
 import catchAsync from "../utils/catchAsync";
+import bcrypt from "bcryptjs";
+import jwt from "jsonwebtoken";
+import dotenv from "dotenv";
 
-const allClients = catchAsync(async (req, res) => {
+dotenv.config();
+const SECRET = process.env.JWT_SECRET;
+
+const allClients = catchAsync(async (req: any, res: any) => {
   const clients = await fetchClients();
 
   res.status(httpStatus.OK).send(clients);
 });
 
-//add client
-
-//hide client
-
 //edit client
+//login client
+const loginClient = catchAsync(async (req: any, res: any) => {
+  const { gov_id, passcode } = req.body;
 
+  if (!gov_id || !passcode) {
+    throw new ApiError(httpStatus.BAD_REQUEST, "Missing data");
+  }
+
+  const client = await getClient(gov_id);
+  if (client.length == 0) {
+    throw new ApiError(httpStatus.BAD_REQUEST, "No client found");
+  }
+  const dbPassword = client[0].passcode;
+  bcrypt.compare(passcode, dbPassword).then((match) => {
+    if (!match) {
+      res.send({ status: "wrong password" });
```

Comment:
add a return here after the response, just so it doesn't reach the second response and get an error